### PR TITLE
feat: add proposal simulator interface function and tests

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #XXXX

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by faddat** [comment]
Once https://github.com/cosmos/ibc-go/pull/4603 or any of the PR's linked to https://github.com/cosmos/ibc-go/issues/4612 are merged to feat/wasm-clients, this PR's tests will pass.

**Review by colin-axner — APPROVED** [review]
ACK pending passing tests

File: modules/light-clients/08-wasm/simulation/proposals.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	}
+}
+
+// SimulateMsgStoreCode returns a random MsgStoreCode for the 08-wasm module
+func SimulateMsgStoreCode(r *rand.Rand, _ sdk.Context, _ []simtypes.Account) sdk.Msg {
+	var signer sdk.AccAddress = address.Module("gov")
+
+	return &types.MsgStoreCode{
+		Signer:       signer.String(),
+		WasmByteCode: []byte{0x01},
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply modules/light-clients/08-wasm/simulation/proposals.go:38]: We could, but it looks like these simulation tests are not really random. Even the SDK tests are just using a pseudo random generator and the test just checks the first output of the generator (which is predictable).

**Review by damiannolan — APPROVED** [review]
🚀

**Pull request merged**